File: src/basic/day05/Question03.java
package basic.day05;

public class Question03 {

	public static class Solution {

		public int solution(int a, int b, int c) {
			int answer = 0;

			double i1 = a + b + c;
			double i2 = Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2);
			double i3 = Math.pow(a, 3) + Math.pow(b, 3) + Math.pow(c, 3);

			if (a != b && a != c && b != c) {
				answer = (int) i1;
			} else if (a == b && a == c) {
				answer = (int) (i1 * i2 * i3);
			} else {
				answer = (int) (i1 * i2);
			}

			return answer;
		}
	}

	public static void main(String[] args) {

		Solution solution = new Solution();

		System.out.println(solution.solution(5, 3, 3));
	}
}

File: src/beginner/day22/Question03.kt
package beginner.day22

import kotlin.math.abs

class Question03{

    fun solution(lines: Array<IntArray>): Int {

        val firstLine = mutableListOf<Int>()
        val secondLine = mutableListOf<Int>()
        val thirdLine = mutableListOf<Int>()

        val firSecLines = mutableSetOf<Int>().toMutableList()
        val secThirdLines = mutableSetOf<Int>().toMutableList()

        for(i in lines.indices){
            for (j in lines[i][0]..lines[i][1]) {
                if (i == 0) firstLine.add(j)
                if (i == 1) secondLine.add(j)
                if (i == 2) thirdLine.add(j)
            }
        }

        for(i in firstLine){
            for(j in secondLine){
                if(i == j) firSecLines.add(i)
            }
        }
        for(i in secondLine){
            for(j in thirdLine){
                if(i == j) secThirdLines.add(i)
            }
        }

        val firstLength = if (firSecLines.size >= 2) (abs(firSecLines[firSecLines.size - 1]) - abs(firSecLines[0])) else 0
        val secondLength = if (secThirdLines.size >= 2) (abs(secThirdLines[secThirdLines.size - 1]) - abs(secThirdLines[0])) else 0

        return firstLength + secondLength
    }
}

fun main() {

    val solution = Question03()

    val lines1 = arrayOf(intArrayOf(0, 1), intArrayOf(2, 5), intArrayOf(3, 9))
    val lines2 = arrayOf(intArrayOf(-1, 1), intArrayOf(1, 3), intArrayOf(3, 9))
    val lines3 = arrayOf(intArrayOf(0, 5), intArrayOf(3, 9), intArrayOf(1, 10))

    val lines4 = arrayOf(intArrayOf(0, 5), intArrayOf(4, 9), intArrayOf(1, 10))

    println(solution.solution(lines1))
    println(solution.solution(lines2))
    println(solution.solution(lines3))

    println(solution.solution(lines4))
}